File: Source/Aladdin.CAPI.PKCS11.AKS/Provider.cs
using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.PKCS11.AKS
{
	///////////////////////////////////////////////////////////////////////////
	// Криптографический провайдер
	///////////////////////////////////////////////////////////////////////////
	public sealed class Provider : ANSI.PKCS11.Provider
	{
		// конструктор
		public Provider() : base("eToken PKCS11 Cryptographic Provider", false) 
        {
            // указать интерфейс вызова функций
            module = Module.Create(new NativeMethods.NativeAPI()); 
        }
        // деструктор
        protected override void OnDispose() 
        {
            // освободить выделенные ресурсы
            RefObject.Release(module); base.OnDispose(); 
        } 
		// интерфейс вызова функций
		public override Module Module { get { return module; }} private Module module;

        // тип структуры передачи параметров механизма PBKDF2
        protected override PBE.PBKDF2.ParametersType PBKDF2ParametersType 
        {
            // тип структуры передачи параметров механизма PBKDF2
            get { return PBE.PBKDF2.ParametersType.ParamsLong; }
        }
    }
}

File: Source/Aladdin.CAPI.PKCS11.AKS.Java/src/aladdin/capi/pkcs11/aks/Provider.java
package aladdin.capi.pkcs11.aks;
import aladdin.pkcs11.*; 
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Криптографический провайдер
///////////////////////////////////////////////////////////////////////////
public class Provider extends aladdin.capi.ansi.pkcs11.Provider
{
    // интерфейс вызова функций
    private final Module module; 
    
	// конструктор
	public Provider(String path) throws IOException 
    { 
        // сохранить переданные параметры
        super("AKS PKCS11 Cryptographic Provider", true);
        
        // открыть модуль
        module = new Module(path); 
    }
	@Override protected void onClose() throws IOException  
    { 
        // освободить выделенные ресурсы
        module.close(); super.onClose();
    } 	
    // интерфейс вызова функций
	@Override public Module module() { return module; } 
    
    // тип структуры передачи параметров механизма PBKDF2
    @Override protected aladdin.capi.pkcs11.pbe.PBKDF2.ParametersType pbkdf2ParametersType() 
    {
        // тип структуры передачи параметров механизма PBKDF2
        return aladdin.capi.pkcs11.pbe.PBKDF2.ParametersType.PARAMS_LONG; 
    }
}
